ilastik/applets/edgeTrainingWithMulticut/edgeTrainingWithMulticutGui.py

```diff
@@ -101,3 +98,27 @@ def configure_gui_from_operator(self, *args):
     def configure_operator_from_gui(self):
         EdgeTrainingGui.configure_operator_from_gui(self)
         MulticutGuiMixin.configure_operator_from_gui(self)
+
+
+def _enable_widget_on_usable_values(widget, *slots):
+    """Enables the widget only if all slots have usable values.
+
+    Subscribe to dirty notifications from the slots and enable
+    the widget only if all slot values are ready and not None.
+    Otherwise, disable the widget.
+
+    Args:
+        widget: Widget to enable or disable.
+        slots: Slots whose values to check.
+
+    Returns:
+        Unsubscribe callable.
+    """
+    def set_widget_enabled(*args, **kwargs):
+        have_values = all(s.ready() and s.value is not None for s in slots)
```

Comment:
for the case of a real connected slot, `slot.value` will request the whole region. Depending on which slots you add here, this might not be desired (say a 100 GB dataset)... But that depends on which slots you connect here - if you are sure that only value slots are passed, then it's fine.

Reply:
Isn't `slot.value` gets used downstream anyway?